eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1729,6 +1722,14 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
         cancelTimer(RevocationTimeout.toString)
       }
 
+      if (state != nextState) {
+        val commitments_opt = nextStateData match {
+          case hasCommitments: HasCommitments => Some(hasCommitments.commitments)
+          case _ => None
+        }
+        context.system.eventStream.publish(ChannelStateChanged(self, nextStateData.channelId, peer, remoteNodeId, state, nextState, commitments_opt))
+      }
+
```

Comment:
There's no need to change where this is done, can you put it back at the beginning of the handler?